Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
As for the case of `:python` if snow leopard + `:python3` scenario, it occurs in `gdal`, `hashpump`, `pyside`, `shiboken`, `sip`, `uwsgi`, and `z3`.